---
title: "Characterization of the multi-omic signature of Minimal Hepatic Encephalopathy in the hippocampus of the *Rattus norvegicus* model induced by diet."
author:
  - "Patricia Checa Gómez"
  - "Ana Clavijo Pizzano"
  - "José Luis del Río Vázquez"
  - "Rocío Cañete Gómez"
format: pdf
fontsize: 9pt
geometry: margin=0.5in
mainfont: "Georgia"
editor_options:
  chunk_output_type: inline
header-includes:
  - \pagenumbering{gobble}
editor: 
  markdown: 
    wrap: sentence
---

#### **1. Description of the data used, objectives, and biological question.**

This project used the HA rats 1 dataset generated by the Neurobiology lab at CIPF (Valencia), processed by BiostatOmics, and further analyzed by the project authors.

The multi-omic data come from *Rattus norvegicus*.
The experimental design aimed to study the molecular and pathophysiological mechanisms of Minimal Hepatic Encephalopathy (MHE), a disease characterized by liver dysfunction leading to high systemic ammonia levels that affect the brain and cause cognitive decline.

MHE was modeled by feeding disease-group rats an ammonia-rich diet and control rats a standard diet for 4 weeks.
Afterward, the hippocampus was extracted—a brain region affected in MHE and other neurodegenerative diseases.

Three omics studies (RNA-Seq, miRNA-Seq, and metabolomics) were performed on hippocampus samples to evaluate MHE’s impact.
Provided data files included gene expression matrices, miRNA expression, metabolite concentrations, lists of differentially expressed genes, and gene-miRNA associations.
Data were normalized and log-transformed.

The project includes one general objective and several specific objectives.

-   **General objective:** characterize the changes at the transcriptome, regulome, and metabolome levels in an integrated manner in the hippocampus of rats with MHE, to allow discrimination between diseased and control animals.

-   **Specific objectives:**

    -   Calculate the desired statistical power and its requirements.

    -   Build a statistical model to identify the variables that best discriminate between control and hyperammonemic rats.

    -   Study cross-regulation among omics and identify potential regulatory candidates.

    -   Visualize the multi-omic data in an integrated manner.

Once this study is completed, the following **biological question** can be addressed:\
Is there a characteristic multi-omic signature in MHE of the *R. norvegicus* model that allows effective discrimination?

#### **2. Chosen integration method and tools used.**

The obtained data have a vertical structure, as all studied omics were obtained from the same set of biological samples.\
Therefore, the main method chosen for this work is statistical integration, for which we will use the R tools MORE (Aguerralde-Martin, M et al., 2023) and MixOmics (Rohart, F et al., 2017), specifically MultiBlock PLS-DA (DIABLO PLS-DA) within MixOmics.\
Additionally, for a more detailed and complementary analysis, a conceptual integration method was selected using the web tool PaintOmics (Liu, T. et al., 2022).

For data preparation, the MultiPower tool (Tarazona, S et al., 2020) was used to calculate statistical power, and Cytoscape software (Franz, M et al., 2023) was used for visualization of regulatory networks.

Below, all necessary libraries for this study are loaded:

```{r}
packages <- c("mixOmics", "FDRsampsize", "lpSolve", "MultiPower", "MORE", "dplyr", "tidyverse")
invisible(suppressMessages(suppressWarnings(lapply(packages, library, character.only = TRUE))))
```

```{r, echo = FALSE}
load("environment_def_1marzo.RData")
```

#### **3. Initial data exploration.**

Next, the raw data will be visualized and studied.\
Regarding transformations, RNA-Seq and miRNA-Seq data have been centered but not scaled, as they come from the same experimental process applied uniformly across all samples.\
On the other hand, metabolomics data have been both centered and scaled due to potentially higher experimental variability and differences in the units of the studied metabolites.

```{r, fig.cap= "Box-and-whisker plots for each omics dataset", include = FALSE}
rnaseq_data <- read.table("datos/RNAseq_data.txt")
mirnaseq_data <- read.table ("datos/miRNAs_data.txt")
metabolomica_data <- read.table ("datos/metabolomic_data.txt")
rnaseq_center <- t (rnaseq_data) # Trasponer matriz
rnaseq_center <- scale (rnaseq_center, center = T, scale = F) # Escalado
par(mfrow=c(1,3), mar=c(8,4,4,2))
boxplot (rnaseq_center [, 1:10], main = "Boxplot - RNA-Seq", las=2, cex.axis=0.7,
         col = "lightsalmon")
mirnaseq_center <- t (mirnaseq_data)
mirnaseq_center <- scale (mirnaseq_center, center = T, scale = F) # Escalado
boxplot (mirnaseq_center [, 1:10], main = "Boxplot - miRNA-Seq", las=2, cex.axis=0.7,
         col = "lightskyblue")
metabolomica_center <- t(metabolomica_data)
metabolomica_center <- scale (metabolomica_center, center = T, scale = T) # Centrado
boxplot (metabolomica_center [, 1:10], main = "Boxplot - Metabolómica", las=2, 
         cex.axis=0.7, col = "lightpink")
```

```{r, fig.cap= "Box-and-whisker plots for each omics dataset", echo=FALSE}
knitr::include_graphics("images/boxplot_iniciales.png")
```

In the box-and-whisker plots (Fig. 1), we can confirm that the data have been centered, as the medians of the distributions are mostly around 0.\
This is because the mean was subtracted from each value, removing the average shift of the data.\
In metabolomics, since scaling affects the variability of each variable, more outliers can be observed compared to the other two omics.\
Additionally, in Appendix 1, the variability explained by each component in each omics dataset can be seen.

#### **4. Calculation of statistical power.**

We used the MultiPower package to calculate the optimal sample size to achieve the desired statistical power.\
The minimum desired power for each omics dataset is 60%, while the overall target power is 80%.\
The cost was defined as unitary due to lack of information about the experimental processes performed and their costs.\
The false discovery rate was set at 5%, and the groups have the same sample size.

```{r, message=F, include = FALSE}
# We create two empty vectors
statdata = statdesign = vector ("list")
# We create a list of omics data. These are the original centered data and the scaled metabolomics data, but we transpose them again because they had been transposed previously.
statdata$rnaseq = t(rnaseq_center)
statdata$mirnaseq = t(mirnaseq_center)
statdata$metabolomica = t(metabolomica_center)
# Design
statdesign$rnaseq = rep (c("C", "H"), each = 8)
statdesign$mirnaseq = rep (c("C", "H"), each = 8)
statdesign$metabolomica = rep (c("C", "H"), each = 8)
miscolores = c("lightsalmon", "lightskyblue", "lightpink")
names (miscolores) = names (statdata)
# Preparing MultiPower
type = rep (2, length(statdata)) # Continuous data
par(mfrow=(c(1,2)))
statResultsEQ = MultiPower (data = statdata, groups = statdesign, type = type,
                            omicCol = miscolores, omicPower = 0.6, 
                            averagePower = 0.8, fdr = 0.05, cost = 1,
                            equalSize = T, max.size = 200)
```

```{r, echo = FALSE}
statResultsEQ$summary
```

```{r, fig.cap = "A. Statistical power VS Sample size. B. Statistical power VS Effect size", echo=FALSE}
knitr::include_graphics("images/graficas_multipower_1.png")
```

We observe that the optimal sample size for the desired power is 144 (Fig. 2A), but we only have 16 samples.\
To improve power, filtering is done using Cohen’s d, which measures the standardized difference between two group means.\
Filtering at a Cohen’s d of 0.9 helps approach the target 80% overall power (Fig. 3A).

```{r, include = FALSE}
par(mfrow=(c(1,2)))
STATpostEQ = postMultiPower(optResults = statResultsEQ, 
                            max.size = 8, omicCol = miscolores)
filteredData = CohenFilter(data = statdata, d = 0.9, # Filter
                           parameters = statResultsEQ$parameters)
sapply(filteredData, dim) # List of filtered variables
```

```{r, fig.cap = "A. Sample size VS Cohen’s d. B. Statistical power at the current and optimal sample size.", echo=FALSE}
knitr::include_graphics("images/graficas_multipower_2.png")
```

#### 5. Statistical data exploration (individual omics analysis, PCA)

An individual analysis was performed for each omics dataset using PCA and sPLS-DA to reduce dimensionality and visualize global variation patterns among samples (PCA), as well as to select the most relevant molecular features that discriminate between groups, helping to understand specific differences in molecular regulation.

##### 5.1 RNA-seq

For this, data previously filtered by Cohen’s d were used:

```{r}
rnaseq <- t(filteredData$rnaseq) 
rnaseq <- scale (rnaseq, scale = F, center = T) 
group <- rep(c("Control", "Hyperammonemia"), each = 8) 
ind <- rownames (rnaseq) 
group.f <- as.factor (group) # Factor group for PLS-DA
```

PCA analysis:

```{r, include = FALSE}
pca_rnaseq <- pca (rnaseq, ncomp = 10, scale = F, center = F) # PCA
plotIndiv (pca_rnaseq, comp = c(1,2),
           ind.names = ind, group = group, cex = 3,
           style = "ggplot2",
           legend = TRUE, legend.position = "right",
           legend.title = NULL, col = c("darkorchid4", "#1874CD"),
           ellipse = TRUE, ellipse.level = 0.95, ellipse.type = "confidence",
           centroid = FALSE, title = "RNAseq")
plotVar (pca_rnaseq, var.names = F, pch = 20, cex = 0.5, comp = 1:2)
plotLoadings (pca_rnaseq, comp = 1, size.name = 0.5, size.title = 1)
```

```{r, fig.cap="RNAseq individuals plot. A. PCA. B. Correlation circle. C. Loadings", echo=FALSE}
knitr::include_graphics("images/imagen_rnaseq.png")
```

The individuals plot (Fig. 4) shows a clear separation between control and hyperammonemic groups, with samples clustering in distinct regions of the principal component space, indicating gene expression changes linked to hyperammonemia.

Most separation occurs along the first component (PC1), which captures the main data variability.
The second component (PC2) does not separate groups.

Loadings on PC1 reveal two gene groups: one positively correlated (right), likely upregulated in hyperammonemia, and one negatively correlated (left), likely upregulated in controls.

This explains the opposite contributions of these genes to group separation.
The main contributing genes to group differences can be identified from the loadings.

Next, PLS-DA analysis was performed on RNA-Seq data.

```{r,  include=FALSE}
set.seed (123)
rna.plsda <- plsda(rnaseq, group.f, ncomp = 8, scale = FALSE)
rna.perf.plsda <- perf (rna.plsda, validation = 'Mfold', folds = 5, 
                        progressBar = F, nrepeat = 10)
plot (rna.perf.plsda, overlay = 'measure', sd = TRUE)
```

```{r, echo=FALSE, fig.cap="BER plot for PLS-DA of RNAseq"}
knitr::include_graphics("images/dist_perf_splsda_rnaseq.png")
```

```{r}
rna.perf.plsda$choice.ncomp
```

As shown in Fig.
5, classification of individuals using different techniques produces no errors starting from the first component, remaining stable thereafter.\
Therefore, two components were chosen for the model, as it is the minimum allowed.

```{r, include=FALSE}
plsda_rnaseq <- plsda(rnaseq, group.f, ncomp = 2, scale = FALSE)
plotIndiv (plsda_rnaseq, comp = 1:2,
           ind.names = ind, group = group, cex = 3,
           style = "ggplot2",
           legend = TRUE, legend.position = "right",
           legend.title = NULL, col = c("darkorchid4", "#1874CD"),
           ellipse = TRUE, ellipse.level = 0.95, ellipse.type = "confidence",
           centroid = FALSE, title = "PLSDA RNAseq")
```

```{r, fig.cap="Individual plot for PLS-DA of RNAseq analysis", echo=FALSE}
knitr::include_graphics("images/plsda_rnaseq.png")
```

Thus, as in the PCA analysis, a clear separation between control and hyperammonemic groups is observed along the first component (Fig. 6).

##### 5.2 miRNA-Seq

```{r}
mirnaseq <- t(filteredData$mirnaseq) # Preparing input
mirnaseq <- scale (mirnaseq, center = T, scale = F) # Centering data
```

A PCA analysis will be performed first:

```{r, include = FALSE}
pca_mirnaseq <- pca (mirnaseq, ncomp = 10, scale = F, center = F) # PCA
plotIndiv (pca_mirnaseq, comp = c(1, 3),
           ind.names = ind, group = group, cex = 3,
           style = "ggplot2",
           legend = TRUE, legend.position = "right",
           legend.title = NULL, col = c("darkorchid4", "#1874CD"),
           ellipse = TRUE, ellipse.level = 0.95, ellipse.type = "confidence",
           centroid = FALSE, title = "miRNAseq")
plotVar (pca_mirnaseq, var.names = F, pch = 20, cex = 0.5, comp = c(1, 3))
plotLoadings (pca_mirnaseq, comp = 1)
```

```{r, fig.cap="miRNAseq individuals plot. A. PCA. B. Correlation circle. C. Loadings", echo=FALSE}
knitr::include_graphics("images/imagen_mirnaseq.png")
```

Fig.
7A shows that components 1 (61% variability) and 3 (10%) best separate the groups.
Separation between hyperammonemic and control samples is less clear than in RNAseq, indicating more heterogeneity in miRNA expression.

Loadings (Fig. 7B) reveal few miRNAs contribute significantly, explaining this heterogeneity.
A dispersed group of loadings in the upper left quadrant corresponds to the hyperammonemic group, suggesting higher expression of those miRNAs.

Compared to RNAseq, miRNAseq shows less variability, consistent with fewer variables

```{r, include=FALSE}
set.seed (123)
mirna.plsda <- plsda(mirnaseq, group.f, ncomp = 9, scale = FALSE)
mirna.perf.plsda <- perf (mirna.plsda, validation = 'Mfold', folds = 5, 
                        progressBar = F, nrepeat = 10)
plot (mirna.perf.plsda, overlay = 'measure', sd = TRUE)
```

```{r, fig.cap="BER plot for miRNAseq PLS-DA", echo=FALSE}
knitr::include_graphics("images/dist_perf_splsda_mirnaseq.png")
```

```{r}
mirna.perf.plsda$choice.ncomp
```

Fig.
8 shows the BER error rate for each component using three distances (maximum, centroids, Mahalanobis).\
The lowest classification error occurs with maximum distance using 3 components, which best classifies the samples.\
Thus, 3 components were used.

```{r, include=FALSE}
plsda_mirnaseq <- plsda(mirnaseq, group.f, ncomp = 3, scale = FALSE)
plotIndiv (plsda_mirnaseq, comp = c(1, 2),
           ind.names = ind, group = group, cex = 3,
           style = "ggplot2",
           legend = TRUE, legend.position = "right",
           legend.title = NULL, col = c("darkorchid4", "#1874CD"),
           ellipse = TRUE, ellipse.level = 0.95, ellipse.type = "confidence",
           centroid = FALSE, title = "PLS-DA miRNAseq")
par (mfrow = c(1, 2))
plotLoadings (plsda_mirnaseq, comp = 1, size.title = 1)
plotLoadings (plsda_mirnaseq, comp = 2, size.title = 1)
```

PLS-DA (Fig. 9A) shows clearer separation of control and hyperammonemic groups using two components, unlike PCA.\
Loadings for each component reveal the same miRNA contributes to group separation in both components (Fig. 9B).

##### 5.3 Metabolomics

For metabolomics, data were scaled and centered as explained in section 3.

```{r}
metabolomica <- t(filteredData$metabolomica) # Preparing data
metabolomica <- scale (metabolomica, scale = T, center = T)
dim (metabolomica)
```

The metabolomics dataset has only 3 variables (columns), so a PCA with 2 principal components will be performed to visualize the data.\
This is due to the prior filtering using Cohen’s d.

```{r, include = FALSE}
pca_metabolomica <- pca (metabolomica, ncomp = 2, scale = F, center = F) # PCA
plotIndiv (pca_metabolomica, comp = 1:2,
           ind.names = ind, group = group, cex = 3,
           style = "ggplot2",
           legend = TRUE, legend.position = "right",
           legend.title = NULL, col = c("darkorchid4", "#1874CD"),
           ellipse = TRUE, ellipse.level = 0.95, ellipse.type = "confidence",
           centroid = FALSE, title = "Metabolomica")
plotLoadings (pca_metabolomica, comp = 1, size.title = 1)
plotLoadings (pca_metabolomica, comp = 2, size.title = 1)
```

```{r, fig.cap="Metabolomics individuals plot. A. PCA. B. Loadings", echo=FALSE}
knitr::include_graphics("images/imagen_metabolomica.png")
```

Fig.
10A shows no clear separation between control and hyperammonemic groups in metabolomics PCA; some control samples overlap with hyperammonemic ones, indicating weak differences.\
Loadings (Fig. 10B) show "PC ae C40:1" contributes most to component 1 variability, and "PC ae C30:1" to component 2.

Despite this, a PLS-DA study was performed.

```{r, include=FALSE}
set.seed (123)
metabolomica.plsda <- plsda (mirnaseq, group.f, ncomp = 3, scale = FALSE)
metabolomica.perf.plsda <- perf (metabolomica.plsda, validation = 'Mfold', 
                                 folds = 5, progressBar = F, nrepeat = 10)
plot (metabolomica.perf.plsda, overlay = 'measure', sd = TRUE)
```

```{r, fig.cap="BER plot for metabolomic PLS-DA", echo=FALSE}
knitr::include_graphics("images/dist_perf_splsda_metabolomica.png")
```

```{r}
metabolomica.perf.plsda$choice.ncomp
```

Fig.
11 shows the lowest BER error with 3 components using maximum distance.\
However, this choice does not reduce dimensionality, as the model still uses many variables for classification rather than simplifying data representation.

```{r, include = FALSE}
plsda_metabolomica <- plsda(metabolomica, group.f, ncomp = 3, scale = FALSE)
plotIndiv(plsda_metabolomica , comp = 1:2,
          group = group, ind.names = ind, cex = 3,
          legend = TRUE, legend.position = "right",
          col = c("darkorchid4", "#1874CD"),
           ellipse = TRUE, ellipse.level = 0.95, ellipse.type = "confidence",
           centroid = FALSE, title = "Metabolómica")
```

```{r, fig.cap="Individual plot para análisis PLS-DA de metabolómica",echo = FALSE}
knitr::include_graphics("images/plsda_metabolomica.png")
```

As shown in Fig.
12, PLS-DA classification reveals no group separation, indicating these metabolites are not suitable variables to distinguish control and hyperammonemia groups.

#### **5. Statistical Integration: MixOmics.**

##### sPLS-DA Multiblock

A multiblock Sparse PLS-DA model was used to efficiently integrate RNAseq, miRNAseq, and metabolomics data, considering correlations without losing data structure.\
Sparse selection identified key interrelated variables to minimize overfitting.\
This descriptive approach aimed to find variables that best distinguish control and hyperammonemic groups, not to build a predictive model due to limited samples.

Data were centered (and scaled for metabolomics) individually, so no further preprocessing was needed before storing them in a list for analysis.

```{r}
# Creamos matriz diseño
design = matrix(1, ncol = length(filteredData), nrow = length(filteredData),
                dimnames = list(names(filteredData), names(filteredData)))
diag(design) = 0; design
filtereddata_centered <- list (rnaseq = rnaseq, mirnaseq = mirnaseq, metabolomica = metabolomica)
```

Next, a *tune.block* was performed to find the optimal number of variables to select per model component.\
The resulting plots identified the best values (in choice.keepX) that maximize group discrimination while reducing dimensionality.\
This tuning improves data interpretation by selecting only the most informative variables in each omics dataset.

```{r}
set.seed(123)
t.filteredData <- lapply(filteredData, t) 
# Create test.keepX as a list with 3 elements, one for each omics dataset
#test.keepX <- list(
#  rnaseq = c(1:10, seq(20, 100, 10)), # There are more than 1000 genes
#  mirnaseq = c(1:10, seq(20, 50, 10)), # There are 39 miRNAs
#  metabolomica = 1:3) # There are just 3 metabolites
# Run tuning for DIABLO (block.sPLS-DA)
#tune.block.splsda <- tune.block.splsda(
#  X = t.filteredData, Y = group.f, ncomp = 3, validation = 'Mfold', nrepeat = 5, 
#  folds = 5, dist = 'mahalanobis.dist', test.keepX = test.keepX, design = design)
#saveRDS(tune.block.splsda, file = "tune_block_splsda.rds")
tune.block.splsda <- readRDS("tune_block_splsda.rds")
```

```{r}
# load ("datos/tune.block.splsda.RData")
tune.block.splsda$choice.keepX
tune.block.splsda$choice.ncomp
```

Recommended to choose one principal component, but at least two are needed to proceed with analysis (e.g., plotVar requires at least two components).
Also, plot the balanced error rate (BER) to justify this choice.

```{r, eval=FALSE}
# Load created plotting function
load ("función_gráfica.RData")
plot.splsda (tune.block.splsda)
```

```{r, echo = FALSE, fig.cap="BER sPLS-DA"}
knitr::include_graphics("images/ber_diablo.png")
```

The graph shows that adding a second component reduces the balanced error rate (BER), especially for RNA-Seq and miRNA-Seq data, improving group discrimination.
Increasing variables could help but is limited by computation time.

miRNAs typically inhibit gene expression, but positive correlations with genes can occur due to indirect effects, like inhibiting repressors.
Such correlations shouldn’t be dismissed without context.

We then select the optimal number of variables and components for a block sPLS-DA model to maximize group separation, and plot loadings to identify key contributing variables.

```{r, include = FALSE}
ncomp = 2
block.splsda = block.splsda(X = t.filteredData, 
                            Y = group.f, 
                            scale = FALSE, ncomp = ncomp, 
                            keepX = tune.block.splsda$choice.keepX, design = design)
plotLoadings(block.splsda, comp = 1, contrib = 'max', block = "metabolomica")
```

Loadings reveal several genes influencing the first component, notably "novel.12125".
Two miRNAs, "rno-miR-6324" and "rno-miR-152-5p", also contribute significantly.
After filtering, the three metabolite variables affect the component more evenly.

MultiBlock sPLS-DA results show no clear separation in the correlation circle plots (Fig. 15A).
However, scatter plots reveal clear grouping of control and hyperammonemic samples in RNA-Seq, but not in miRNA-Seq or metabolomics (Fig. 15B).

```{r, include=FALSE}
plotVar(block.splsda, style = 'graphics', legend = TRUE, var.names = F)
plotIndiv(block.splsda, ind.names = FALSE, legend = TRUE)
```

```{r, fig.cap="sPLS-DA MultiBlock plot. A. Correlation circle. B. Scatter plots", echo=FALSE}
knitr::include_graphics("images/graficas_block.png")
```

In the following Diablo plots, scatter plots for each block are shown (Fig. 16A).\
RNA-seq displays a clear separation between groups, indicating strong discriminative power.\
miRNA-seq also separates groups, though less distinctly, while metabolomics shows the least separation.\
Correlation values between components reveal the highest correlation (0.62) between RNA-seq and miRNA-seq, and the lowest (0.12) between metabolomics and miRNA-seq.

A circosPlot (Fig. 16B) shows inter-omics variable correlations, highlighting the miRNA "rno-miR-6324", which is linked to many genes (mostly inhibitory) and two metabolites—though such metabolite links require further study.\
Positive regulation with two genes may reflect indirect effects, e.g., via repressor degradation.

```{r, include = FALSE}
plotDiablo(block.splsda, ncomp = 1)
circosPlot(block.splsda, comp = 1, cutoff = 0.7, size.variables = 0.60)
```

```{r, fig.cap="A. Scatter plots by block and correlation. B. CircosPlot", echo=FALSE}
knitr::include_graphics("images/diablo_correlation_circos.png")
```

Finally, the MultiBlock sPLS-DA model was validated.\
The goal was not predictive applicability to other datasets, as the sample size and study characteristics prevent a precise model.\
It was trained and tested using "Leave One Out" due to the small sample size.\
The lowest error rate was in miRNA-seq, suggesting it would have the best predictive capacity under better study conditions.\
All predictions were correct except one, though this result should be interpreted cautiously.\
The model is more useful for identifying variables most characteristic of hyperammonemia than for prediction.

```{r}
set.seed(123)
diablo_perf <- perf(block.splsda, 
                    validation = "loo",
                    progressBar = FALSE)
diablo_perf$error.rate 
diablo_perf$AveragedPredict.class
```

#### **6. Statistical integration: MORE.**

We chose to implement the MORE model in this study to identify significant regulators (miRNAs) of gene expression using a multiple linear regression framework.
The input gene expression data are the RNA-Seq measurements after filtering by Cohen's d.
When the model was tested using data filtered by differentially expressed genes, no substantial results were obtained, so we opted to use the previously mentioned (Cohen's d-filtered) set to provide more information to the model.

Additionally, we selected the multiple linear regression model with elastic net regularization ("MLR+EN", method = 'MLR').

First, we again use the gene expression (rnaseq) and miRNA (mirnaseq) datasets already filtered by Cohen's d that were loaded earlier.
The file with the target genes for each miRNA (associations_data) is then loaded:

```{r}
associations_data <- read.table ("datos/targets_mRNA_gene.txt", header = T)
```

A table is created with the experimental design and experimental conditions:

```{r}
samples <- paste0(rep(c("C", "H"), each = 8), 1:8) # Condiciones 
condition <- ifelse (grepl ("^C", samples), "control", "hiperammonemic") 
expdesign <- data.frame (row.names = samples, condition = condition)
```

A minimum change vector (value 0) is set, which the regulator (in this case, the miRNAs) must display under different conditions to be included in the model.

```{r}
minVar <- 0 
names (minVar) <- "miRNAseq"
```

From the mirnaseq data, regulators also present in the associations_data are selected, and all columns are converted to character type to ensure the data are interpreted as text:

```{r}
mirnaseq <- mirnaseq[rownames(mirnaseq) %in% associations_data$miRNA, ]
associations_data <- associations_data %>% select(target_gene, miRNA) %>% mutate(across(everything(), as.character))
```

Next, the MORE function is executed and a summary of significant regulations for each gene is generated.

```{r echo=TRUE, message=FALSE, warning=FALSE, eval = FALSE}
MLRmodel = more(targetData = rnaseq, regulatoryData = list(miRNA = mirnaseq),
                associations = list(miRNA = associations_data), 
                condition = expdesign, omicType = 0, scaleType = 'auto', 
                interactions = TRUE, minVariation = minVar, varSel = 'EN', 
                alfaEN = NULL, correlation = 0.7, method = 'MLR', parallel = FALSE) 
regpcond = RegulationPerCondition(MLRmodel) 
summaryPlot(MLRmodel, regpcond, byTargetF = FALSE, filterR2 = 0.6)
saveRDS(MLRmodel, file = "mlrmodel.rds")
networkMORE(regpcond, cytoscape = TRUE)
```

The regulation plot shows differences between groups, indicating altered regulatory mechanisms between conditions (Fig. 17A).\
However, the highest R-squared imposed here is 0.6, with a low percentage of significant data in both cases.\
This may be due to the study’s low power, yet a slight difference between conditions is visible after applying R-squared; more relevant regulations appear in the "hyperammonemic" condition, suggesting gene regulation activation by miRNAs.

From the MORE model summary, three regulators are deemed "important," regulating at least 10 genes each: "rno-miR-3065-3p," "rno-miR-3557-5p," and "novel_229."\
This regulator acts differently between conditions; under the hyperammonemic diet, gene expression decreases while the regulator’s expression slightly increases.

A differential regulatory network between conditions inferred by the model is visualized in Cytoscape (Fig. 17B).\
The miRNA-gene regulation network is shown.\
Most miRNAs regulate multiple genes, and most genes are regulated by a single miRNA.\
The three key regulators mentioned earlier are visible, with rno-miR-3065-3p regulating the highest number of genes.\
Overall, coefficients increase in the hyperammonemic condition, both positively and negatively, as the legend indicates.

##### ORA (Over Representation Analysis)

For this analysis, we use rat annotation obtained from Biomart.

```{r}
annotation <- read.table("datos/annotation_rat_correct.txt", 
                         header = TRUE, sep = "\t")
annotation <- as.matrix (annotation) 
colnames(annotation) <- c("genes", "GO terms", "description") 
regincond_harats = RegulationInCondition(regpcond, cond = 'Group_hiperammonemic') 
ORA_harats = oraMORE(MLRmodel, regincond_harats, byHubs = TRUE, annotation = annotation, 
                     alpha = 0.05, p.adjust.method = "fdr") 
```

Considering the 3 main regulators or hubs in ORA, no enriched GO terms were found for "rno-miR-3065-3p."\
For "rno-miR-3557-5p," 33 terms were identified, and for "novel_229," 51 terms.

Overall, the enriched pathways mainly relate to cell signaling and apoptosis regulation, suggesting these key regulators modulate survival and cell death processes.
Immune response and energy metabolism pathways are also highlighted.

#### 7. Conceptual integration: PaintOmics.

Functional enrichment analysis helps interpret multi-omics data to understand molecular mechanisms in hyperammonemia.\
PaintOmics4 was used to integrate gene expression, miRNAs, and metabolites filtered by Cohen’s d \> 0.9, applying Kendall correlation (-0.5 threshold) for regulators.\
743 metabolic and functional pathways (355 KEGG, 388 Reactome) were identified, with 17 statistically significant.\
KEGG showed equal predominance of metabolic processes and human diseases (\~26% each), linking altered metabolism and pathogenesis.\
Reactome emphasized signal transduction processes (\~18%), including tyrosine kinase receptor signaling, followed by metabolic pathways (\~16%).

The multi-omic network revealed PI3K-AKT as a central hub, associated with multiple pathological processes including cancers.\
Metabolite analysis identified L-kynurenine (p = 0.01160) as a central hub metabolite involved in tryptophan metabolism and linked to neuroinflammation and neurodegeneration.\
Significant pathway enrichments included non-small cell lung cancer, starch and sucrose metabolism, and B cell receptor signaling, suggesting broad alterations in energy metabolism and cell signaling in the hyperammonemic rat hippocampus.

These results align with prior studies linking sustained PI3K-AKT activation to neuroinflammation and cognitive decline in hyperammonemia, impacting key processes like cell survival and synaptic plasticity.\
L-kynurenine’s hub role suggests NF-κB activation may induce pro-inflammatory mediators (TNFα, HMGB1), diverting tryptophan metabolism via IDO enzyme induction stimulated by cytokines (TNFα, IFN-γ) in the hippocampus.

#### 8. Discussion on applying other methods.

-   MOFA could identify shared factors across omics layers, aiding variability interpretation, but MultiBlock sPLS-DA already integrated layers effectively, making MOFA unnecessary here.\
-   WGCNA suits detecting gene co-expression modules linked to metabolites, but since this study focused on multi-omics integration for sample classification, WGCNA was less suitable.\
-   CustOmics, based on deep learning, needs larger, more complex datasets; given this study’s scale and focus on interpretability, deep learning was not ideal.

Though these methods are theoretically applicable, MultiBlock sPLS-DA offered effective data integration.
MOFA and WGCNA depend on specific goals, while CustOmics requires larger data volumes.

#### 9. Conclusions.

This multi-omics integrative analysis characterized transcriptome, regulome, and metabolome changes in hippocampus of MHE rat models, differentiating Control vs. HA groups.\
RNA-seq and miRNA-seq showed the strongest discriminative power, with clear group separation in PCA and sPLS-DA analyses.\
Significant gene and miRNA expression changes suggest strong molecular regulation linked to hyperammonemia.\
Metabolomics showed alterations in key metabolites but lower discriminative ability, indicating possible post-transcriptional modulation or need for higher analytic resolution.

The MultiBlock sPLS-DA (DIABLO) model identified variables best distinguishing control and hyperammonemic rats, showing strong correlation between RNA-seq and miRNA-seq discrimination, with metabolomics less aligned.\
The model was validated with low balanced error rates, supporting robustness.\
Cross-omics regulation analyses identified key regulators (rno-miR-6324, rno-miR-3065-3p, rno-miR-3557-5p, novel_229) potentially central in gene expression modulation under hyperammonemia.

In summary, a clear multi-omics signature exists in MHE in *R. norvegicus*, enabling effective group discrimination.\
DIABLO confirmed RNA-seq and miRNA-seq combination provides clear separation and high correlation between discriminant components.\
MORE identified key regulators possibly modulating observed changes.

These findings underscore the importance of multi-omics approaches in characterizing complex diseases, allowing not only discrimination between pathophysiological states but also identification of regulatory mechanisms.\
This multi-omics signature could be a foundational step toward understanding molecular alterations in MHE and guiding future validation and clinical applications.

#### 10. References

Aguerralde-Martin, M., Tomás-Riquelme, B., Clemente-Císcar, M., Conesa, A., & Tarazona, S.
(2023).
MORE: Multi-Omics Regulation.
R package version 1.0.

Arenas, Y. M., & Felipo, V. (2023).
Sustained hyperammonemia activates NF-ΚB in purkinje neurons through activation of the TRKB-PI3K-AKT pathway by Microglia-Derived BDNF in a RAT model of minimal hepatic encephalopathy.
Molecular Neurobiology, 60(6), 3071–3085.
<https://doi.org/10.1007/s12035-023-03264-4>

Argelaguet, R., Velten, B., Arnol, D., Dietrich, S., Zenz, T., Marioni, J. C., Buettner, F., Huber, W., & Stegle, O.
(2018).
Multi-Omics Factor Analysis—a framework for unsupervised integration of multi-omics data sets.
Molecular Systems Biology, 14(6).
<https://doi.org/10.15252/msb.20178124>

Benkirane, H., Pradat, Y., Michiels, S., & Cournède, P. (2022).
CustOmics: A versatile deep-learning based strategy for multi-omics integration.
arXiv (Cornell University).
<https://doi.org/10.48550/arxiv.2209.05485>

Franz, M., Lopes, C. T., Fong, D., Kucera, M., Cheung, M., Siper, M. C., Huck, G., Dong, Y., Sumer, O., & Bader, G. D.
(2023).
Cytoscape.js 2023 update: a graph theory library for visualization and analysis.
*Bioinformatics, 39*(1), btad031.
<https://doi.org/10.1093/bioinformatics/btad031>

Langfelder, P., & Horvath, S.
(2008).
WGCNA: an R package for weighted correlation network analysis.
BMC Bioinformatics, 9(1).
<https://doi.org/10.1186/1471-2105-9-559>

Liu, T., Salguero, P., Petek, M., Martinez-Mira, C., Balzano-Nogueira, L., Ramšak, Ž., McIntyre, L., Gruden, K., Tarazona, S., & Conesa, A.
(2022).
PaintOmics 4: new tools for the integrative analysis of multi-omics datasets supported by multiple pathway databases.
Nucleic Acids Research, 50(W1), W551–W559.
<https://doi.org/10.1093/nar/gkac352>

Rohart, F., Gautier, B., Singh, A., & Lê Cao, K.-A.
(2017).
mixOmics: An R package for 'omics feature selection and multiple data integration.
PLOS Computational Biology, 13(11), e1005752.
<https://doi.org/10.1371/journal.pcbi.1005752>

Tarazona, S., Balzano-Nogueira, L., Gómez-Cabrero, D., et al. (2020).
Harmonization of quality metrics and power calculation in multi-omic studies.
Nature Communications, 11, 3092.
<https://doi.org/10.1038/s41467-020-16937-8>

Wickham, H., Averick, M., Bryan, J., Chang, W., McGowan, L. D., François, R., Grolemund, G., Hayes, A., Henry, L., Hester, J., Kuhn, M., Pedersen, T. L., Miller, E., Bache, S. M., Müller, K., Ooms, J., Robinson, D., Seidel, D. P., Spinu, V., Takahashi, K., Vaughan, D., Wilke, C., Woo, K., & Yutani, H.
(2019).
Welcome to the tidyverse.
Journal of Open Source Software, 4(43), 1686.
<https://doi.org/10.21105/joss.01686>

Wickham, H., François, R., Henry, L., & Müller, K.
(2023).
dplyr: A Grammar of Data Manipulation.
R package version 1.1.4.
<https://dplyr.tidyverse.org>
